downgrade.jl

```diff
@@ -191,6 +492,94 @@ for dir in dirs
                 See the errors above for which packages need their bounds adjusted.
                 """)
         end
-        @info "All forcedeps checks passed for $dir"
+
+        # Check test project (excluding local source packages)
+        if !check_forced_lower_bounds(test_project_file, main_manifest, forcedeps_ignore)
+            error("""
+                forcedeps check failed: Some test dependencies did not resolve to their lower bounds.
+
+                See the errors above for which packages need their bounds adjusted.
+                """)
+        end
+
+        @info "All forcedeps checks passed for merged project"
+    end
+
+    # Process any remaining directories that aren't main or test
+    other_dirs = filter(d -> d != "." && d != "test", dirs)
+    for dir in other_dirs
```

Comment:
This whole `for` loop is the same as below in the `else` branch. This can be modularized to avoid code duplication.